FAQ.md

```diff
@@ -3,502 +3,565 @@ title: TiDB FAQ
-    Create_routine_priv   ENUM('N','Y') NOT NULL DEFAULT 'N',
-    Alter_routine_priv    ENUM('N','Y') NOT NULL DEFAULT 'N',
-    Index_priv      ENUM('N','Y') NOT NULL DEFAULT 'N',
-    Create_user_priv    ENUM('N','Y') NOT NULL DEFAULT 'N',
-    Event_priv      ENUM('N','Y') NOT NULL DEFAULT 'N',
-    Trigger_priv      ENUM('N','Y') NOT NULL DEFAULT 'N',
-    PRIMARY KEY (Host, User));
-
-INSERT INTO mysql.user VALUES ("%", "root", "", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y", "Y");
-
-```
-#### 怎样把数据从 TiDB 导出／导入呢？
-TiDB 遵从 MySQL 协议，可以使用 Mydumper 工具导出数据，使用 Loader 工具导入数据，和 MySQL 导出／导入方式一样。
-
-#### 数据删除后空间多长时间空间回收？
-Delete，Truncate 和 Drop 都不会立即释放空间，对于 Truncate 和 Drop 操作，在达到 TiDB 的 GC (Garbage Collection) 时间后（默认 10 分钟），TiDB 的 GC 机制会删除数据并释放空间。对于 Delete 操作 TiDB 的 GC 机制会删除数据，但不会释放空间，而是当后续数据写入 RocksDB 且进行 Compact 时对空间重新利用。
+### 3.4.10 TiKV 是否可以使用 SAS/SATA 盘或者进行 SSD/SAS 混合部署？
 
-#### 数据删除后查询速度变慢了？
-大量删除数据后，会有很多无用的 key 存在，影响查询效率。目前正在开发 Region Merge 功能，完善之后可以解决这个问题，具体看参考[最佳实践](https://pingcap.com/blog-cn/tidb-best-practice/)中的删除数据部分。
+不可以使用，TiDB 在进行 OLTP 场景中，数据访问和操作需要高 IO 磁盘的支持，TiDB 作为强一致的分布式数据库，存在一定的写放大，如副本复制、存储底层 Compaction，因此，TiDB 部署的最佳实践中推荐用户使用 NVMe SSD 磁盘作为数据存储磁盘。另外，TiKV 与 PD 不能混合部署。
 
-#### 数据删除最高效最快的方式？
-在删除大量数据的时候，建议使用 `Delete * from t where xx limit 5000`（xx 建议在满足业务过滤逻辑下，尽量加上强过滤索引列或者直接使用主键选定范围，如 `id >= 5000*n+m and id < 5000*(n+1)+m `）这样的方案，通过循环来删除，用 `Affected Rows == 0` 作为循环结束条件，这样避免遇到事务大小的限制。如果一次删除的数据量非常大，这种循环的方式会越来越慢，因为每次删除都是从前向后遍历，前面的删除之后，短时间内会残留不少删除标记（后续会被 GC 掉），影响后面的 Delete 语句。如果有可能，建议把 Where 条件细化。可以参考官网[最佳实践](https://pingcap.com/blog-cn/tidb-best-practice/)。
+### 3.4.11 数据表 Key 的 Range 范围划分是在数据接入之前就已经划分好了吗？
 
-#### Syncer / Drainer 同步出错、中断怎么处理？
-+ 查看 log 分析出错原因。
-+ 查看 [Syncer](https://pingcap.com/docs-cn/tools/syncer/)/ [Drainer](https://pingcap.com/docs-cn/tools/tidb-binlog-kafka/#pump-drainer-%E9%85%8D%E7%BD%AE) 文档寻求帮助。
-+ 及时更换新版的 Syncer / Drainer。
+不是的，这个和 MySQL 分表规则不一样，需要提前设置好，TiKV 是根据 Region 的大小动态分裂的。
 
-### 性能调优
+### 3.4.12 Region 是如何进行分裂的？
 
-#### TiDB 如何提高数据加载速度？
+首先不是前期划分好的，但确实有 Region 分裂机制，有一个参数 `region_split_size`，超过这个值就会触发分裂，分裂后的信息会汇报给 PD。
 
-主要两个方面：
+### 3.4.13 TiKV 是否有类似 MySQL 的 `innodb_flush_log_trx_commit` 参数，来保证提交数据不丢失？
 
-+ 目前正在开发分布式导入工具 Lightning，需要注意的是数据导入过程中为了性能考虑，不会执行完整的事务流程，所以没办法保证导入过程中正在导入的数据的 ACID 约束，只能保证整个导入过程结束以后导入数据的 ACID 约束。因此适用场景主要为新数据的导入（比如新的表或者新的索引），或者是全量的备份恢复（先 Truncate 原表再导入）。
-+ TiDB 的数据加载与磁盘以及整体集群状态相关，加载数据时应关注该主机的磁盘利用率/ TiClient Error / Backoff / Thread CPU 等相关 metric，可以分析相应瓶颈。
+是的，TiKV 单机的存储引擎目前使用两个 rockdb 实例，其中一个存储 raft-log，TiKV 有个 sync-log 参数，在ture 的情况下，每次提交都会强制刷盘到 raft-log，如果发生 crash 后，通过 raft-log 进行 KV 数据的恢复。
 
+### 3.4.14 对 WAL 存储有什么推荐的硬件配置，例如 SSD，RAID 级别，RAID 卡 cache 策略，NUMA 设置 ,文件系统选择，操作系统的 IO 调度策略等？
 
-### 备份恢复
+WAL 属于顺序写，目前我们并没有单独对他进行配置，建议 SSD，RAID 如果允许的话，最好是 RAID 10，RAID 卡 cache、操作系统 IO 调度目前没有针对性的 best pratise，Linux 7 以上默认配置即可，NUMA 没有特别建议，Muma 内存分配策略可以尝试使用 `interleave = all`，文件系统建议 EXT4。
```

Comment:
EXT4 -> ext4
Change the capitalization format to [the frequently used version](https://en.wikipedia.org/wiki/Ext4) here.